### PR TITLE
task/modify quantity - 11897

### PR DESCRIPTION
**Pull request by Vadim-x-squad**
https://user-images.githubusercontent.com/127092693/228224275-2c81057d-4afc-44d4-9218-4eb70873da3c.mp4

### CODE REVIEW

**Review by ducdetronquito — COMMENTED** [review]
Hey @Vadim-x-squad, would you mind to provide a gif/video of the result ?

**Comment by vadimsharanov** [comment]
> Hey @Vadim-x-squad, would you mind to provide a gif/video of the result ?

Yes, i forgot to make i disabled case, let me push changes and the video will be there after!

File: example/src/ModifyQuantity/ModifyQuantityPage.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    const theme = useTheme();
+
+    const styles = StyleSheet.create({
+        container: {
+            alignItems: 'center',
+            marginTop: theme.sw.spacing.m,
+        },
+    });
+
+    const [value, setValue] = useState<string>('0');
```
Comment (ducdetronquito) [inline example/src/ModifyQuantity/ModifyQuantityPage.tsx:15]: I think we should use an int here: it better convey that this is a quantity and not a textual input.
This input will be used to modify a quantity, and the quantity will be stored as an int anyway.

Comment (Vadim-x-squad) [reply example/src/ModifyQuantity/ModifyQuantityPage.tsx:15]: I will check that :)

File: example/src/ModifyQuantity/ModifyQuantityPage.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    };
+    const onMinus = () => {
+        setValue((parseInt(value) - 1).toString());
+    };
+
+    return (
+        <Screen style={styles.container}>
+            <ModifyQuantity onMinus={onMinus} onAdd={onAdd} text={'Stick'} inputValue={value} />
+            <ModifyQuantity
+                disabled
```
Comment (ducdetronquito) [inline example/src/ModifyQuantity/ModifyQuantityPage.tsx:28]: I think the API should not expose the disabled props, but a `minValue` and `maxValue` that controls when "plus" or "minus" actions should be disabled.

```tsx
const [value, setValue] = useState<number>(0);
<ModifyQuantity
    minValue={0}
    maxValue={100}
    value={value}
    onChange={value => setValue(value)}
    text={'Stick'}
    icon={'minus-fill'} />



interface Props {
    text: string;
    onAdd: () => void;
    disabled?: boolean;
    onMinus: () => void;
    style?: ViewStyle;
    inputValue: string;
}
export const ModifyQuantity = () => {
   const minusDisabled = value <= minValue;
    // ...
                <Pressable disabled={minusDisabled } hitSlop={8} onPress={onChange(value -1)}>
                    <Icon
                        size={28}
                        color={minusDisabled  ? '#919EAB3D' : theme.sw.colors.neutral[800]}
                        name={icon}
                    />
                </Pressable>

    // And same for plus
}  
```

Comment (Vadim-x-squad) [reply example/src/ModifyQuantity/ModifyQuantityPage.tsx]: updated!

File: example/src/ModifyQuantity/ModifyQuantityPage.tsx (hunk 1 above)
Comment (ducdetronquito) [reply example/src/ModifyQuantity/ModifyQuantityPage.tsx:15]: Convert it to string when you pass it to the TextInput value props

```ts
<TextInput value={parseInt(value)}/>
```

File: example/src/ModifyQuantity/ModifyQuantityPage.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const styles = StyleSheet.create({
+        container: {
+            alignItems: 'center',
+            marginTop: theme.sw.spacing.m,
+        },
+    });
+
+    const [value, setValue] = useState<number>(0);
+
+    const onAdd = () => {
```
Comment (ducdetronquito) [inline example/src/ModifyQuantity/ModifyQuantityPage.tsx:17]: I think the logic when we increment or decrement the value should be kept within the ModifyQuantity component, the caller only needs to know when the value has changed.

Therefore, I would rather see a `onValueChange` property that takes a callback with the changed value in parameters ``<ModifyQuantity onChange={(value) => setValue(value)}` />

File: src/components/modifyQuantity/ModifyQuantity.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        inputContainer: {
+            alignItems: 'center',
+            flexDirection: 'row',
+        },
+        input: {
+            marginBottom: 0,
+            marginHorizontal: theme.sw.spacing.l,
+        },
+        inputText: {
+            fontSize: 28,
```
Comment (ducdetronquito) [inline src/components/modifyQuantity/ModifyQuantity.tsx:64]: I see 16px on the design system no ?
cf: https://www.figma.com/file/MD0xdHA7E93GKAwkSRpduK/SW-mobile-app---Design-Kit?node-id=984-1415&t=3kNwwPqELMX05L3S-4

**Review by ducdetronquito — COMMENTED** [review]
Some minor comments, but the PR looks good !

File: example/src/ModifyQuantity/ModifyQuantityPage.tsx (hunk 3 above)
Comment (Vadim-x-squad) [reply example/src/ModifyQuantity/ModifyQuantityPage.tsx]: updated!

**Pull request merged**